### ISSUE TITLE
GeoProximityLocations -- proximity's p is in uppercase but should be in lowercase.

### ISSUE THREAD

**Issue by hinling-blisspoint** [hashicorp/terraform-cdk#2056]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

```
cdktf debug
language: typescript
cdktf-cli: 0.12.1
node: v16.16.0
cdktf: 0.12.1
constructs: 10.1.85
jsii: null
terraform: 1.2.7
arch: arm64
os: darwin 21.6.0
```

### Affected Resource(s)

DataAwsRoute53TrafficPolicyDocumentRule

### Debug Output

https://gist.github.com/hinling-blisspoint/daa5cdf98523f2a7c57356c856207fbd

### Expected Behavior

created the traffic policy with geoproximity rule:
expected policy doc from aws cli:
```
{
  "AWSPolicyFormatVersion": "2015-10-01",
  "RecordType": "CNAME",
  "Endpoints": {
    "endpoint-geoproximity-PMXS": {
      "Type": "value",
      "Value": "d-redacted.execute-api.us-west-2.amazonaws.com"
    },
    "endpoint-geoproximity-wcvy": {
      "Type": "value",
      "Value": "d-redacted.execute-api.us-east-1.amazonaws.com"
    }
  },
  "Rules": {
    "geoproximity-start-ycsO": {
      "RuleType": "geoproximity",
      "GeoproximityLocations": [
        {
          "EndpointReference": "endpoint-geoproximity-PMXS",
          "Bias": 0,
          "Region": "aws:route53:us-west-2",
          "EvaluateTargetHealth": true,
          "HealthCheck": "redacted"
        },
        {
          "Bias": 0,
          "Region": "aws:route53:us-east-1",
          "EvaluateTargetHealth": false,
          "HealthCheck": "redacted",
          "EndpointReference": "endpoint-geoproximity-wcvy"
        }
      ]
    }
  },
  "StartRule": "geoproximity-start-ycsO"
}
```

### Actual Behavior

```
Error: error creating Route53 Traffic Policy (ghost-bidding-dev.infra-bpm.link): InvalidTrafficPolicyDocument: The document contains unexpected or invalid elements
               │        status code: 400, request id: 7fdfedd7-12fb-4438-a621-2f73ca85e816
```

actual policy document from log:
```
"{\"AWSPolicyFormatVersion\":\"2015-10-01\",\"RecordType\":\"CNAME\",\"StartRule\":\"geoproximity-start-dev\",\"Endpoints\":{\"ghost-api-gateway-endpoint-dev-us-east-1\":{\"Type\":\"value\",\"Value\":\"d-redacted.execute-api.us-east-1.amazonaws.com\"},\"ghost-api-gateway-endpoint-dev-us-west-2\":{\"Type\":\"value\",\"Value\":\"d-redacted.execute-api.us-west-2.amazonaws.com\"}},\"Rules\":{\"geoproximity-start-dev\":{\"RuleType\":\"geoproximity\",\"GeoProximityLocations\":[{\"EndpointReference\":\"ghost-api-gateway-endpoint-dev-us-east-1\",\"Region\":\"aws:route53:us-east-1\",\"Bias\":\"0\"},{\"EndpointReference\":\"ghost-api-gateway-endpoint-dev-us-west-2\",\"Region\":\"aws:route53:us-west-2\",\"Bias\":\"0\"}]}}}",
```

### Steps to Reproduce
1. create a DataAwsRoute53TrafficPolicyDocument with rule type geoproximity:
```
    const routeZone = new DataAwsRoute53Zone(this, 'infra-bpm-link-zone', {
      name: 'somedomain.test'
    });

    const trafficPolicyDocument = new DataAwsRoute53TrafficPolicyDocument(this, 'some-traffic-policy-doc', {
      recordType: 'CNAME',
      endpoint: generateEndpointArray(stateMap, config.environment),
      startRule: `geoproximity-start-${config.environment}`,
      rule: [{
        id: `geoproximity-start-${config.environment}`,
        type: 'geoproximity',
        geoProximityLocation: [{
          bias: "0",
          region: 'aws:route53:us-west-2',
          endpointReference: `some-api-gateway-endpoint-${config.environment}-us-west-2`,
          evaluateTargetHealth: false
        },{
          bias: "0",
          region: `aws:route53:us-east-1`,
          endpointReference: `some-api-gateway-endpoint-${config.environment}-us-east-1`,
          evaluateTargetHealth: false
        }]
      }]
    });

    const trafficPolicy = new Route53TrafficPolicy(this, 'ghost-geoprox-traffic-policy', {
      name: CommonStack.globalDomainName,
      comment: 'ghost-bidding traffic policy',
      document: trafficPolicyDocument.json
    });
```
2. Run `cdktf deploy` to actually apply the plan. 

### References

aws documentation about geoproximity policy syntax:  https://docs.aws.amazon.com/Route53/latest/APIReference/api-policies-traffic-policy-document-format.html#traffic-policy-document-format-rules-geoproximity

**Comment by ansgarm** [comment]
Hi @hinling-blisspoint,

it seems that this is an upstream issue in the Terraform AWS provider. I had a quick look at the code their and it seems that the uppercase `P` stems from this line: https://github.com/hashicorp/terraform-provider-aws/blob/2ca401f41b6ad2b578498e923b81c0b18a74cc9b/internal/service/route53/traffic_policy_document_data_source.go#L361

**Comment by ansgarm** [comment]
Raised this with the Terraform AWS Provider team and will report back if my suspicion turns out to be true (or not).

**Comment by hinling-blisspoint** [comment]
thanks @ansgarm . not a huge deal. for now, to work around this, I just inline the policy document in the `Route53TrafficPolicy`.

**Comment by ewbankkit** [comment]
Yes, it seems like the capitalization here
https://github.com/hashicorp/terraform-provider-aws/blob/8184d586643b8dcd35c14e9cb00f2882b9810ee5/internal/service/route53/traffic_policy_document_model.go#L40
is incorrect - should be `GeoproximityLocations`.

**Comment by xiehan** [comment]
Looks like the upstream issue was solved in hashicorp/terraform-provider-aws#27473 so I'll go ahead and close this.
